docs/tutorials/data_provider.md

```diff
@@ -269,31 +269,41 @@ use tinystr::tinystr;
 
 pub struct CustomDecimalSymbolsProvider<P>(P);
 
-impl<P> AnyProvider for CustomDecimalSymbolsProvider<P>
+fn transform(any_res: AnyResponse) -> Result<AnyResponse, DataError> {
+    let mut dec_res: DataResponse<DecimalSymbolsV1Marker> = any_res.downcast()?;
```

Comment:
Note that for some reason when I had this function inlined below, I got the following compile error:

```
error[E0271]: type mismatch resolving `<DecimalSymbolsV1Marker as DataMarker>::Yokeable == <M as DataMarker>::Yokeable`
  --> src/lib.rs:301:77
   |
37 |             let mut dec_res: DataResponse<DecimalSymbolsV1Marker> = any_res.downcast()?;
   |                                                                             ^^^^^^^^ expected `DecimalSymbolsV1<'_>`, found associated type
   |
   = note:       expected struct `DecimalSymbolsV1<'static>`
           found associated type `<M as DataMarker>::Yokeable`
   = help: consider constraining the associated type `<M as DataMarker>::Yokeable` to `DecimalSymbolsV1<'static>`
   = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
```

Moving the code into a helper function fixes the error. Compiler bug?

Reply:
Hmm, still seems like a bug that `let foo: Foo<Bar> = foo()?` would resolve differently than `let foo = foo::<Bar>()` with the signature `fn foo<T>() -> Result<Foo<T>>`